### PR TITLE
SERVICES-1081 index marketplace transactions too

### PR DESCRIPTION
**Pull request by johnykes**
- Index marketplace transactions too (not only events)

### CODE REVIEW

File: src/db/marketplaces/marketplace-events.repository.ts [hunk 1]
```diff
@@ -32,16 +32,27 @@ export class MarketplaceEventsRepository extends Repository<MarketplaceEventsEnt
     return savedRecords;
   }
```
Comment (danielailie) [inline src/db/marketplaces/marketplace-events.repository.ts:34]: could you change the name, it is not clear why you get be marketpace and Timestamp

File: src/modules/marketplaces/marketplaces.elastic.service.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  RangeLowerThan,
+  RangeGreaterThan,
+  ElasticSortOrder,
+  QueryConditionOptions,
+} from '@elrondnetwork/erdnest';
+import { Injectable } from '@nestjs/common';
+import { constants } from 'src/config';
+import { MarketplaceEventsIndexingRequest } from './models/MarketplaceEventsIndexingRequest';
+
+@Injectable()
```
Comment (danielailie) [inline src/modules/marketplaces/marketplaces.elastic.service.ts:13]: why is this a service, you could do something similar to nft-traits.elastic.queries.ts

**Review by danielailie — CHANGES_REQUESTED** [review]

File: src/modules/marketplaces/marketplaces.elastic.service.ts (hunk 2 above)
Comment (johnykes) [reply src/modules/marketplaces/marketplaces.elastic.service.ts]: Right, fixed

File: src/db/marketplaces/marketplace-events.repository.ts (hunk 1 above)
Comment (johnykes) [reply src/db/marketplaces/marketplace-events.repository.ts:34]: Fixed

File: src/modules/marketplaces/marketplaces-events-indexing.service.ts [hunk 3]
```diff
@@ -85,7 +81,10 @@ export class MarketplaceEventsIndexingService {
         throw new Error(`beforeTimestamp can't be less than afterTimestamp`);
       }
 
-      const [newestTimestamp] = await this.getEventsAndSaveToDb(input);
+      let newestTimestamp: number;
+
```
Comment (danielailie) [inline src/modules/marketplaces/marketplaces-events-indexing.service.ts:85]: why here you rewrite the newestTimestamp? Are there moments when await this.getTxAndSaveToDb(input); returns null for timestamp?

**Review by danielailie — CHANGES_REQUESTED** [review]

File: src/modules/marketplaces/marketplaces-events-indexing.service.ts (hunk 3 above)
Comment (johnykes) [reply src/modules/marketplaces/marketplaces-events-indexing.service.ts:88]: Fixed

File: src/modules/marketplaces/marketplaces-events-indexing.service.ts [hunk 4]
```diff
@@ -107,30 +106,55 @@ export class MarketplaceEventsIndexingService {
...
+          transactions,
+          input.marketplaceAddress,
+        );
+
+        if (input.stopIfDuplicates && savedItemsCount !== totalEventsCount) {
+          return false;
+        }
+      },
+    );
+
```
[the remark this answers is not in the document]
Comment (johnykes) [reply src/modules/marketplaces/marketplaces-events-indexing.service.ts]: Removed. was used for console.logs :)

**Review by CatalinFaurPaul — APPROVED** [review]

**Pull request merged**